packages/ra-core/src/types.ts

```diff
@@ -67,9 +67,17 @@ export type AuthProvider = {
     checkError: (error: any) => Promise<void>;
     getIdentity?: () => Promise<UserIdentity>;
     getPermissions: (params: any) => Promise<any>;
+    handleCallback?: <
+        HandleCallbackResult = AuthRedirectResult
+    >() => Promise<HandleCallbackResult | void>;
```

Comment:
Why do you need a generic type here?

Besides, I believe you should type the error too (error type was `void | { redirectTo?: string, logoutOnFailure?: boolean, message?: string }` if I'm not mistaken).

Reply:
> Why do you need a generic type here?

Do avoid adding `any` as we did on `login`

Reply:
Btw, this also means I was wrong about adding `message` to `AuthRedirectResult`, I think we souldn't